File: backend/src/modules/providers/service.model.ts
import {
    Schema,
    model,
} from "mongoose";

import {
    ServiceDocument,
} from "./service.types";

import {
    ServiceCategory,
} from "./service.enums";

const serviceSchema =
    new Schema<ServiceDocument>(
        {
            title: {
                type: String,

                required: true,

                trim: true,

                minlength: 3,

                maxlength: 100,

                index: true,
            },

            description: {
                type: String,

                required: true,

                trim: true,

                minlength: 10,

                maxlength: 2000,
            },

            price: {
                type: Number,

                required: true,

                min: 0,
            },

            duration: {
                type: Number,

                required: true,

                min: 15,
            },

            category: {
                type: String,

                enum: Object.values(
                    ServiceCategory
                ),

                required: true,

                index: true,
            },

            imageUrl: {
                type: String,

                trim: true,
            },

            capacityPerSlot: {
                type: Number,

                required: true,

                min: 1,
            },

            isActive: {
                type: Boolean,

                default: true,

                index: true,
            },

            createdBy: {
                type:
                    Schema.Types.ObjectId,

                ref: "User",

                required: true,
            },
        },
        {
            timestamps: true,

            versionKey: false,
        }
    );

serviceSchema.index({
    title: "text",

    description: "text",
});
export const ServiceModel =
    model<ServiceDocument>(
        "Service",
        serviceSchema
    );

File: mobile/mentecart/lib/features/services/models/service_model.dart
class ServiceModel {
  final String id;

  final String title;

  final String description;

  final double price;

  final int duration;

  final String category;

  final String imageUrl;

  final int capacityPerSlot;

  final bool isActive;

  ServiceModel({
    required this.id,
    required this.title,
    required this.description,
    required this.price,
    required this.duration,
    required this.category,
    required this.imageUrl,
    required this.capacityPerSlot,
    required this.isActive,
  });

  factory ServiceModel.fromJson(Map<String, dynamic> json) {
    return ServiceModel(
      id: json['id'],

      title: json['title'],

      description: json['description'],

      price: (json['price'] as num).toDouble(),

      duration: json['duration'],

      category: json['category'],

      imageUrl: json['imageUrl'],

      capacityPerSlot: json['capacityPerSlot'],

      isActive: json['isActive'],
    );
  }
}
